packages/uiweb/src/lib/components/space/SpaceBanner/SpaceBanner.tsx

```diff
@@ -1,15 +1,322 @@
 import React from 'react';
-import { useSpaceData } from '../../../hooks';
+import styled from 'styled-components';
+
+import live from './assets/live.svg';
+import scheduled from './assets/CalendarBlank.svg';
+
+import { ISpacesTheme } from '../theme';
+import { ThemeContext } from '../theme/ThemeProvider';
+import { useSpaceData } from './../../../hooks';
+import { getDateAndTime, getStatus } from './utils';
+
+import * as PushAPI from '@pushprotocol/restapi';
 
 export interface ISpaceBannerProps {
-  // Add props specific to the SpaceBanner component
-  temporary?: string; // just to remove eslint error of empty prop
+  spaceId: string;
+  orientation?: 'maximized' | 'minimized';
 }
 
-export const SpaceBanner: React.FC<ISpaceBannerProps> = () => {
+/**
+ * @interface IThemeProps
+ * this interface is used for defining the props for styled components
+ */
+interface IThemeProps {
+  theme?: ISpacesTheme;
+  orientation?: string;
+  status?: string;
+}
+
+export const SpaceBanner: React.FC<ISpaceBannerProps> = ({
+  spaceId,
+  orientation,
+}) => {
+
+  const theme = React.useContext(ThemeContext);
+  // const [spaceBannerData, setSpaceBannerData] = React.useState<any>();
   const { spaceBannerData, setSpaceBannerData } = useSpaceData();
 
-  // Use spaceBannerData and setSpaceBannerData in your component
+  React.useEffect(() => {
```

Comment:
let's make a hook for getting spaceData, that will take input as spaceId or anything that's required and will check if there is data present already in spaceContext, then return that otherwise call the api, set the data in context and also return that

Reply:
and how do we implement updating the cache?

Reply:
updating the cache only means to call the setSpacesInfo with updated info